// helpers/api/items/index.js
const { default: axios } = require('axios');

const login = (baseURL, { email, password }) => axios.post(`${baseURL}/api/auth/login`, {
    email,
    password,
});

async function createItem(baseURL, { email, password }, { name, description, sku }) {
    const { headers } = await login(baseURL, { email, password, });

    // Create a new item
    const { data } = await axios.post(
        `${baseURL}/api/items`,
        { name, description, sku },
        {
            headers: {
                cookie: headers["set-cookie"][0],
            },
        }
    );

    return data;
}

async function deleteItem(baseURL, { email, password }, uuid) {
    const { headers } = await login(baseURL, { email, password, });

    // Delete item
    await axios.delete(
        `${baseURL}/api/items/${uuid}`,
        {
            headers: {
                cookie: headers["set-cookie"][0],
            },
        }
    );
}

async function createWarehouse(baseURL, { email, password }, { zipCode, city, province }) {
    const { headers } = await login(baseURL, { email, password, });

    // Create a new warehouse
    const { data } = await axios.post(
        `${baseURL}/api/warehouses`,
        { zipCode, city, province },
        {
            headers: {
                cookie: headers["set-cookie"][0],
            },
        }
    );

    return data;
}

async function deleteWarehouse(baseURL, { email, password }, uuid) {
    const { headers } = await login(baseURL, { email, password, });

    // Delete warehouse
    await axios.delete(
        `${baseURL}/api/warehouses/${uuid}`,
        {
            headers: {
                cookie: headers["set-cookie"][0],
            },
        }
    );
}

module.exports = {
    createItem,
    deleteItem,
    createWarehouse,
    deleteWarehouse,
}
